cunumeric/module.py

```diff
@@ -1392,6 +1399,233 @@ def squeeze(a: ndarray, axis: Optional[NdShapeLike] = None) -> ndarray:
     return a.squeeze(axis=axis)
 
 
+def broadcast_shapes(
+    *args: Union[NdShapeLike, Sequence[NdShapeLike]]
+) -> NdShape:
+    """
+
+    Broadcast the input shapes into a single shape.
+
+    Parameters
+    ----------
+    `*args` : tuples of ints, or ints
+        The shapes to be broadcast against each other.
+
+    Returns
+    -------
+    tuple : Broadcasted shape.
+
+    See Also
+    --------
+    numpy.broadcast_shapes
+
+    Availability
+    --------
+    Single CPU
+
+    """
+    return _broadcast_shapes(*args)
+
+
+def _broadcast_to(
+    arr: ndarray,
+    shape: NdShapeLike,
+    subok: bool = False,
+    broadcasted: bool = False,
+) -> ndarray:
+    # create an array object w/ options passed from 'broadcast' routines
+    arr = array(arr, copy=False, subok=subok)
+    # 'broadcast_to' returns a read-only view of the original array
+    out_shape = broadcast_shapes(arr.shape, shape)
+    result = ndarray(
+        shape=out_shape,
+        thunk=arr._thunk.broadcast_to(out_shape),
+        flags=arr.flags,
+    )
+    result.setflags(write=False)
+    return result
+
+
+@add_boilerplate("arr")
+def broadcast_to(
+    arr: ndarray, shape: NdShapeLike, subok: bool = False
+) -> ndarray:
+
+    """
+
+    Broadcast an array to a new shape.
+
+    Parameters
+    ----------
+    arr : array_like
+        The array to broadcast.
+    shape : tuple or int
+        The shape of the desired array.
+        A single integer i is interpreted as (i,).
+    subok : bool, optional
+        If True, then sub-classes will be passed-through,
+        otherwise the returned array will be forced to
+        be a base-class array (default).
+
+    values : array_like
+        These values are appended to a copy of arr.
+        It must be of the correct
+        shape (the same shape as arr, excluding axis).
+        If axis is not specified, values can be any shape
+        and will be flattened before use.
+    axis : int, optional
+        The axis along which values are appended.
+        If axis is not given, both `arr` and `values`
+        are flattened before use.
+
+    Returns
+    -------
+    broadcast : array
+        A readonly view on the original array with the given shape.
+        It is typically not contiguous.
+        Furthermore, more than one element of a broadcasted array
+        may refer to a single memory location.
+
+    See Also
+    --------
+    numpy.broadcast_to
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+
+    """
+    return _broadcast_to(arr, shape, subok)
+
+
+def _broadcast_arrays(
+    arrs: list[ndarray],
+    subok: bool = False,
+) -> list[ndarray]:
+    # create an arry object w/ options passed from 'broadcast' routines
+    arrays = list(array(arr, copy=False, subok=subok) for arr in arrs)
```

Comment:
```suggestion
    arrays = [array(arr, copy=False, subok=subok) for arr in arrs]
```
list comprehensions are almost always preferable in cases like this, they dispatch immediately to the CPython level (are slightly more performant)